src/libMessage/Messenger.cpp

```diff
@@ -2710,22 +2844,40 @@ bool Messenger::GetLookupGetDSBlockFromSeed(const vector<unsigned char>& src,
   return true;
 }
 
-bool Messenger::SetLookupSetDSBlockFromSeed(vector<unsigned char>& dst,
-                                            const unsigned int offset,
-                                            const uint64_t lowBlockNum,
-                                            const uint64_t highBlockNum,
-                                            const vector<DSBlock>& dsBlocks) {
+bool Messenger::SetLookupSetDSBlockFromSeed(
+    vector<unsigned char>& dst, const unsigned int offset,
+    const uint64_t lowBlockNum, const uint64_t highBlockNum,
+    const std::pair<PrivKey, PubKey>& lookupKey,
+    const vector<DSBlock>& dsBlocks) {
   LOG_MARKER();
 
   LookupSetDSBlockFromSeed result;
 
   result.set_lowblocknum(lowBlockNum);
   result.set_highblocknum(highBlockNum);
 
+  SerializableToProtobufByteArray(lookupKey.second, *result.mutable_pubkey());
+
   for (const auto& dsblock : dsBlocks) {
     DSBlockToProtobuf(dsblock, *result.add_dsblocks());
   }
 
+  Signature signature;
+  if (result.dsblocks().size() > 0) {
+    // Serialize the transactions to temp buffer, later use it to calculate the
```

Comment:
Comment is wrong. Please check the other parts of the code too for similar changes.